### PR TITLE
Add support for Chef icons

### PR DESCRIPTION
**Pull request by cnunciato**
This adds support for including the Chef icons @jamescott has been working on. The Icon component page contains initial documentation and usage examples. Please enjoy.

The thinking here, workflow-wise, is that SVG source files, exported from Illustrator, will be put through a cleanup and compilation process (a new Grunt task uses [svgo](https://github.com/svg/svgo) and [svgstore](https://github.com/FWeinb/grunt-svgstore) to handle that) and then brought into the library at `lib/assets/images/icons`; "raw" (i.e., Illustrator-exported/intermediate) SVGs will be left out of the repo. 

Icons may be referenced using a class or symbol name that maps directly to the source filename, which I don't love &mdash; e.g., things look like this:

    <i class="icon_objects_server_organization"></i>

.. instead of this:

    <i class="chef-organization"></i>

.. but I expect we'll be able to come up with something better here, so the naming will probably change.

This model (SVG sprite+symbol) should also apply well to the Chef logo.

### CODE REVIEW

File: lib/assets/stylesheets/chef/components/icon.scss [hunk 1]
```diff
@@ -5,17 +5,281 @@ name: icon
...
+  <i class="icon_objects_server_environment"></i>
+  <i class="icon_objects_server_group"></i>
+  <i class="icon_objects_server_node"></i>
+  <i class="icon_objects_server_organization"></i>
+  <i class="icon_objects_server_policy"></i>
+  <i class="icon_objects_server_roles"></i>
+  <i class="icon_objects_server_run_list"></i>
+  <i class="icon_objects_server_search"></i>
+  <i class="icon_objects_server_user"></i>
+  <i class="icon_objects_server_workstation"></i>
```
Comment (smith) [inline lib/assets/stylesheets/chef/components/icon.scss]: Would it be too much of a pain to put `title` attributes on these so you could hover over them to see what they mean?

File: lib/assets/javascripts/chef/components/icon.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+//= require ../chef
+
+(function($, window) {
+  'use strict';
+
+  Chef.Web.Core.components.icon = {
+    init: function() {
```
Comment (smith) [inline lib/assets/javascripts/chef/components/icon.js]: So, if you're using this method, and add new `<i class="..."></i>` icons to the DOM, they would not show up unless you called `Chef.Web.Core.components.icon.init()` again. We might want to note this somewhere.

File: lib/assets/stylesheets/chef/components/icon.scss (hunk 1 above)
Comment (cnunciato) [inline lib/assets/stylesheets/chef/components/icon.scss]: Not at all, I agree that'd be nicer, yeah.

**Comment by smith** [comment]
:+1: but having titles and my note about the JS would be great too.

File: lib/assets/javascripts/chef/components/icon.js (hunk 2 above)
Comment (cnunciato) [inline lib/assets/javascripts/chef/components/icon.js]: Ah, that's true, I overlooked that. I'll add a note and think about how to make this better.

**Pull request merged**